README.md

```diff
@@ -25,10 +27,52 @@ module "mylb" {
 }
 ```
 
+Test
+-----
+### Configurations
+- [Configure Terraform for Azure](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/terraform-install-configure)
+
+We provide 2 ways to build, run, and test module on local dev box:
+
+### Native(Mac/Linux)
+
+#### Prerequisites
+- [Ruby **(~> 2.3)**](https://www.ruby-lang.org/en/downloads/)
+- [Bundler **(~> 1.15)**](https://bundler.io/)
+- [Terraform **(~> 0.11.0)**](https://www.terraform.io/downloads.html)
+
+#### Environment setup
+We provide simple script to quickly set up module development environment:
+```sh
+$ curl -sSL https://raw.githubusercontent.com/Azure/terramodtest/master/tool/env_setup.sh | sudo bash
+```
+#### Run test
+Then simply run it in local shell:
+```sh
+$ bundle install
+$ rake build
+```
+
+### Docker
+We provide Dockerfile to build and run module development environment locally:
+
+#### Prerequisites
+- [Docker](https://www.docker.com/community-edition#/download)
+
+#### Build the image
+```sh
+docker build -t azure-loadbalancer .
+```
+#### Run test
+```sh
```

Comment:
IMO we should point people to this repo on how to build/test with the Docker option until there is a NEED to have them custom build their own image. https://github.com/azure/terraform-test.  You can then simply use this for the test section: https://github.com/Azure/terraform-azurerm-computegroup/blob/master/README.md#run-test

Reply:
This section is meant to show external developers how to build a local image to test their modules on. We are also attempting to keep the instructions the same across all module repo's, [network module](https://github.com/Azure/terraform-azurerm-network) for example.

Reply:
Agreed it would be best to be consistent. I believe we should give people the simplest approach that works for the Docker steps. The module you reference could also use the simpler approach with docker.